### ISSUE TITLE
Installation via nix fails

### ISSUE THREAD

**Issue by capitanbatata** [input-output-hk/stack2nix#72]
I've tried installing `stack2nix` by running:

```sh
nix-env -i stack2nix
```

but I get this error:

```text
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/c6vddkijjfigf72dw9md3gg0q2gafx21-stack2nix-0.1.3.0 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --with-gcc=gcc --package-db=/tmp/nix-build-stack2nix-0.1.3.0.drv-0/package.conf.d --ghc-option=-j4 --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --disable-coverage --enable-library-vanilla --disable-executable-dynamic --enable-tests --ghc-option=-split-sections
Configuring stack2nix-0.1.3.0...
Setup: Encountered missing dependencies:
Glob >=0.7.14 && <0.8, data-fix ==0.0.4
builder for ‘/nix/store/df3iilpp6rysjhcl8w2xy5ij8v54b9h3-stack2nix-0.1.3.0.drv’ failed with exit code 1
error: build of ‘/nix/store/df3iilpp6rysjhcl8w2xy5ij8v54b9h3-stack2nix-0.1.3.0.drv’ failed
```

**Comment by capitanbatata** [comment]
Related to #71.

**Comment by domenkozar** [comment]
What nixpkgs version are you using?

**Comment by capitanbatata** [comment]
I don't remember doing anything special to select a `nixpkgs` repository. How can I give you this information?

**Comment by domenkozar** [comment]
could you try https://github.com/input-output-hk/stack2nix/pull/73

**Comment by domenkozar** [comment]
Fixed in #73

**Comment by capitanbatata** [comment]
This is weird. After pulling from `master` and performing a `stack clean` and `stack build` I'm still running into the same issue.

```text
stack-1.5.1: configure (lib + exe)
stack-1.5.1: build (lib + exe)
trifecta-1.6.2.1: configure
stack-1.5.1: copy/register
Progress: 2/4
--  While building package trifecta-1.6.2.1 using:
      /home/damian/.stack/programs/x86_64-linux/ghc-nopie-8.0.2/bin/ghc --make -odir /tmp/stack24371/trifecta-1.6.2.1/.stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/setup -hidir /tmp/stack24371/trifecta-1.6.2.1/.stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/setup -i -i. -package=Cabal-2.0.0.2 -clear-package-db -global-package-db -package-db=/home/damian/.stack/snapshots/x86_64-linux-nopie/lts-9.11/8.0.2/pkgdb /tmp/stack24371/trifecta-1.6.2.1/Setup.lhs /home/damian/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs -main-is StackSetupShim.mainOverride -o /tmp/stack24371/trifecta-1.6.2.1/.stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/setup/setup -threaded
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/damian/Documents/github/stack2nix/.stack-work/logs/trifecta-1.6.2.1.log

    [1 of 2] Compiling Main             ( /tmp/stack24371/trifecta-1.6.2.1/Setup.lhs, /tmp/stack24371/trifecta-1.6.2.1/.stack-work/dist/x86_64-linux-nopie/Cabal-2.0.0.2/setup/Main.o )
    
    /tmp/stack24371/trifecta-1.6.2.1/Setup.lhs:7:31: error:
        Module
        ‘Distribution.Package’
        does not export
        ‘PackageName(PackageName)’
```

I don't know whether the stack version or something else differs on my system. But for now I'm not using `stack2nix` so I might give this another try in the future. Thanks anyway.

**Comment by domenkozar** [comment]
@capitanbatata could you try the branch in https://github.com/input-output-hk/stack2nix/pull/75/files

**Comment by capitanbatata** [comment]
Sure, I'm building it right now. Let's see...

**Comment by capitanbatata** [comment]
Same error, `trifecta` biting me in the behind again :(

**Comment by capitanbatata** [comment]
BTW, if I'm the only one experiencing this issue please feel free to leave it closed. Also let me know if I can provide any more information. All in all, I won't be working with Nix in the foreseeable future, so this issue is not affecting me. Thanks!

**Comment by domenkozar** [comment]
@capitanbatata `cabal --version` should be 2.0

**Comment by capitanbatata** [comment]
mmm, the version in `~/.local/bin` (installed by `stack` presumably) is `1.24.0.2`. But I thought stack used the right cabal version as specified in the `stack2nix.cabal` file.

**Comment by domenkozar** [comment]
Sadly not, we require `cabal-install` to be 2.0, can you make sure it's available and try again?

**Comment by capitanbatata** [comment]
I'm trying installing Cabal 2.0 in the `stack2nix` directory, with

```sh
cd stack2nix
stack install Cabal-2.0 
```
I would prefer not to change the cabal version globally, since it might break my other projects.

I do get the following warning:

```text
WARNING: /home/damian/Documents/github/stack2nix/.stack-work/downloaded/3O3rwxxL92z8/cabal2nix.cabal was generated with a newer version of hpack, please upgrade and try again.
```

Is there a recommended version of `hpack` I should use as well?

I don't know how I can install cabal-2.0 locally for `stack2nix`rm

**Comment by capitanbatata** [comment]
Ok, I've learned something new, doing `stack install <package>` will simply install the package into `~./local/bin`. So I have `cabal-install` 2.0 for all my projects now :scream_cat:. Anyway, after this upgrade I still get the same error. I don't know whether the `hpack` warning has anything to do with it. 

```text
hpack  --version
hpack version 0.18.1
```

**Comment by domenkozar** [comment]
@capitanbatata from logs I see you're not passing `--nix`. Can you try with `stack build --nix`?

**Comment by capitanbatata** [comment]
No I didn't. Let me try again with this flag.

**Comment by capitanbatata** [comment]
Yes, building with `--nix` fixes the problem. Thanks.
